tools/lsvast/lsvast.cpp

```diff
@@ -122,162 +225,297 @@ std::ostream& operator<<(std::ostream& out, const vast::fbs::uuid::v0* uuid) {
-    std::cout << "(error reading index file " << path.str() << ")\n";
+    std::cout << indent << "(error reading index file " << path.str() << ")\n";
   }
   switch (index->index_type()) {
     case vast::fbs::index::Index::v0:
-      print_index_v0(index->index_as_v0(), indent);
+      print_index_v0(index->index_as_v0(), indent, formatting);
       break;
     default:
       std::cout << "(unknown partition version)\n";
   }
 }
 
-void print_segment_v0(const vast::fbs::segment::v0* segment, int indent = 0) {
+void print_segment_v0(const vast::fbs::segment::v0* segment,
+                      indentation& indent,
+                      const formatting_options& formatting) {
   vast::uuid id;
   if (segment->uuid())
     unpack(*segment->uuid(), id);
-  std::string spaces(indent, ' ');
-  std::cout << spaces << "Segment\n";
-  spaces += "  ";
-  std::cout << spaces << "uuid: " << to_string(id) << "\n";
-  std::cout << spaces << "events: " << segment->events() << "\n";
+  std::cout << indent << "Segment\n";
+  indented_scope _(indent);
+  std::cout << indent << "uuid: " << to_string(id) << "\n";
+  std::cout << indent << "events: " << segment->events() << "\n";
+
+  if (formatting.verbosity >= output_verbosity::verbose) {
+    std::cout << indent << "table_slices:\n";
+    indented_scope _(indent);
+    size_t total_size = 0;
+    for (auto slice : *segment->slices()) {
+      auto table_slice = slice->data_nested_root();
+      vast::record_type layout;
+      vast::fbs::deserialize_bytes(table_slice->layout(), layout);
+      std::cout << indent << layout.name() << ": " << table_slice->rows()
+                << " rows";
+      if (formatting.print_bytesizes) {
+        auto size = slice->data()->size();
+        std::cout << " (" << print_bytesize(size, formatting) << ")";
+        total_size += size;
+      }
+      std::cout << '\n';
+    }
+    if (formatting.print_bytesizes)
+      std::cout << indent << "total: " << print_bytesize(total_size, formatting)
+                << "\n";
+  }
 }
 
-void print_segment(vast::path path, int indent = 0) {
+void print_segment(vast::path path, indentation& indent,
+                   const formatting_options& formatting) {
   auto segment = read_flatbuffer_file<vast::fbs::Segment>(path);
   if (!segment) {
     std::cout << "(error reading segment file " << path.str() << ")\n";
   }
   switch (segment->segment_type()) {
     case vast::fbs::segment::Segment::v0:
-      print_segment_v0(segment->segment_as_v0(), indent);
+      print_segment_v0(segment->segment_as_v0(), indent, formatting);
       break;
     default:
       std::cout << "(unknown partition version)\n";
   }
 }
 
 int main(int argc, char** argv) {
-  if (argc != 2) {
-    std::cerr << "Usage: ./lsvast <path/to/vast.db>\n";
+  std::string raw_path;
+  struct formatting_options format;
+  format.print_bytesizes = true;
+  format.verbosity = output_verbosity::verbose;
+  for (int i = 1; i < argc; ++i) {
+    auto arg = argv[i];
```

Comment:
```suggestion
    auto arg = std::string_view{argv[i]};
```

This may make the subsequent comparisons less C-ish.